From 8017d8def9fa6d5409936ac9452e847fd0e42913 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 20 Nov 2013 23:51:16 +0000
Subject: [PATCH 3/5] More Python stuff (7eafc9)

Search for python2 then python26 then finally python
as python3 is sometimes python and that doesn't work
---
 .../head/100-Use-AC_PROG_PATHS-for-Python.patch    | 62 +++++++++++++++++-----
 1 file changed, 49 insertions(+), 13 deletions(-)

diff --git a/patches/llvm/head/100-Use-AC_PROG_PATHS-for-Python.patch b/patches/llvm/head/100-Use-AC_PROG_PATHS-for-Python.patch
index fea75bb..e85d24d 100644
--- a/patches/llvm/head/100-Use-AC_PROG_PATHS-for-Python.patch
+++ b/patches/llvm/head/100-Use-AC_PROG_PATHS-for-Python.patch
@@ -1,18 +1,23 @@
-From daaa53c8a12ca650d2ecfbc8a4911bb634c56afc Mon Sep 17 00:00:00 2001
+From a268da0f777f547b5093d0c38a53a1eeabf58271 Mon Sep 17 00:00:00 2001
 From: Ray Donnelly <mingw.android@gmail.com>
 Date: Sat, 16 Nov 2013 02:06:23 +0000
-Subject: [PATCH 1/5] Use AC_PROG_PATHS for Python
+Subject: [PATCH] Use AC_PROG_PATHS for Python
 
 Was using AC_PROG_PATH incorrectly and
 failing to find python2 if python wasn't
-present
+present.
+
+Also re-ordered the search list order to
+[python26 python2 python] so that systems
+where python is python3, but with python2
+also installed can be used.
 ---
  autoconf/configure.ac |  2 +-
- configure             | 19 ++++++++++++-------
- 2 files changed, 13 insertions(+), 8 deletions(-)
+ configure             | 29 +++++++++++++++++++----------
+ 2 files changed, 20 insertions(+), 11 deletions(-)
 
 diff --git a/autoconf/configure.ac b/autoconf/configure.ac
-index 5678a21..a447d24 100644
+index 8d23ad3..fa9ab32 100644
 --- a/autoconf/configure.ac
 +++ b/autoconf/configure.ac
 @@ -1371,7 +1371,7 @@ else
@@ -20,28 +25,59 @@ index 5678a21..a447d24 100644
    fi
  
 -  AC_PATH_PROG([PYTHON], [python python2 python26],
-+  AC_PATH_PROGS([PYTHON], [python python2 python26],
++  AC_PATH_PROGS([PYTHON], [python2 python26 python],
                 [AC_MSG_RESULT([not found])
                  AC_MSG_ERROR([could not find python 2.5 or higher])])
  fi
 diff --git a/configure b/configure
-index 2fa5682..ad63002 100755
+index e977d6c..32aa404 100755
 --- a/configure
 +++ b/configure
-@@ -11766,8 +11766,10 @@ else
+@@ -8825,7 +8825,9 @@ if test "${enable_ltdl_install+set}" = set; then
+ fi
+ 
+ 
+- if test x"${enable_ltdl_install-no}" != xno; then
++
++
++if test x"${enable_ltdl_install-no}" != xno; then
+   INSTALL_LTDL_TRUE=
+   INSTALL_LTDL_FALSE='#'
+ else
+@@ -8833,7 +8835,9 @@ else
+   INSTALL_LTDL_FALSE=
+ fi
+ 
+- if test x"${enable_ltdl_convenience-no}" != xno; then
++
++
++if test x"${enable_ltdl_convenience-no}" != xno; then
+   CONVENIENCE_LTDL_TRUE=
+   CONVENIENCE_LTDL_FALSE='#'
+ else
+@@ -10582,7 +10586,7 @@ else
+   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
+   lt_status=$lt_dlunknown
+   cat > conftest.$ac_ext <<EOF
+-#line 10585 "configure"
++#line 10589 "configure"
+ #include "confdefs.h"
+ 
+ #if HAVE_DLFCN_H
+@@ -11762,8 +11766,10 @@ else
  echo "$as_me: WARNING: specified python ($PYTHON) is not usable, searching path" >&2;}
    fi
  
 -  # Extract the first word of "python python2 python26", so it can be a program name with args.
 -set dummy python python2 python26; ac_word=$2
-+  for ac_prog in python python2 python26
++  for ac_prog in python2 python26 python
 +do
 +  # Extract the first word of "$ac_prog", so it can be a program name with args.
 +set dummy $ac_prog; ac_word=$2
  { echo "$as_me:$LINENO: checking for $ac_word" >&5
  echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
  if test "${ac_cv_path_PYTHON+set}" = set; then
-@@ -11793,11 +11795,6 @@ done
+@@ -11789,11 +11795,6 @@ done
  done
  IFS=$as_save_IFS
  
@@ -53,7 +89,7 @@ index 2fa5682..ad63002 100755
    ;;
  esac
  fi
-@@ -11811,6 +11808,14 @@ echo "${ECHO_T}no" >&6; }
+@@ -11807,6 +11808,14 @@ echo "${ECHO_T}no" >&6; }
  fi
  
  
@@ -69,5 +105,5 @@ index 2fa5682..ad63002 100755
  
  { echo "$as_me:$LINENO: checking for python >= 2.5" >&5
 -- 
-1.8.5.rc2
+1.8.4.2
 
-- 
1.8.4.2

